software/tracksight/frontend/components/widgets/Widget.tsx

```diff
@@ -0,0 +1,22 @@
+"use client";
+
+import getWidgetSchema from "@/lib/getWidgetSchema";
+import WidgetRenderer, { WIDGET_TYPE } from "@/lib/types/Widget";
+
+const Widget: WidgetRenderer<WIDGET_TYPE> = (props) => {
+  const { type } = props;
+
+  const widgetSchema = getWidgetSchema(type);
+
+  const Renderer = widgetSchema.renderer;
+
+  if (!Renderer) {
+    console.error(`Attempted to render widget with non-existant type '${type}'`);
+
+    return;
+  }
```

Comment:
extending on previous comment, probably best to do a switch case on the discriminated union and dispatch the renderer here, than configure the renderer in some other file. I think colocating all the renderers here is the best play. Generally I try to avoid dynamic dispatch as much as possible because it makes code very hard to reason about